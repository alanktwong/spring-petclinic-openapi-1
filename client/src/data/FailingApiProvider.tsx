import * as React from 'react';
import { FailingApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithFailingApiProps {
  failingApi: FailingApi;
}

export const FailingApiContext = React.createContext(new FailingApi());

export const withFailingApi = <P extends WithFailingApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithFailingApi extends React.Component<
    Subtract<P, WithFailingApiProps>
  > {
    static contextType = FailingApiContext;
    context!: React.ContextType<typeof FailingApiContext>;

    render() {
      return <Component failingApi={this.context} {...(this.props as P)} />;
    }
  };
