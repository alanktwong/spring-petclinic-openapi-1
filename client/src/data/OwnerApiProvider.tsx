import * as React from 'react';
import { OwnerApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithOwnerApiProps {
  ownerApi: OwnerApi;
}

export const OwnerApiContext = React.createContext(new OwnerApi());

export const withOwnerApi = <P extends WithOwnerApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithOwnerApi extends React.Component<Subtract<P, WithOwnerApiProps>> {
    static contextType = OwnerApiContext;
    context!: React.ContextType<typeof OwnerApiContext>;

    render() {
      return <Component ownerApi={this.context} {...(this.props as P)} />;
    }
  };
