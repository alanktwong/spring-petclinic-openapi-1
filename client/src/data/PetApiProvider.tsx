import * as React from 'react';
import { PetApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithPetApiProps {
  petApi: PetApi;
}

export const PetApiContext = React.createContext(new PetApi());

export const withPetApi = <P extends WithPetApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithPetApi extends React.Component<Subtract<P, WithPetApiProps>> {
    static contextType = PetApiContext;
    context!: React.ContextType<typeof PetApiContext>;

    render() {
      return <Component petApi={this.context} {...(this.props as P)} />;
    }
  };
