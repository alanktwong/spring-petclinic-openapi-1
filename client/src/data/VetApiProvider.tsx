import * as React from 'react';
import { VetApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithVetApiProps {
  vetApi: VetApi;
}

export const VetApiContext = React.createContext(new VetApi());

export const withVetApi = <P extends WithVetApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithVetApi extends React.Component<Subtract<P, WithVetApiProps>> {
    static contextType = VetApiContext;
    context!: React.ContextType<typeof VetApiContext>;

    render() {
      return <Component vetApi={this.context} {...(this.props as P)} />;
    }
  };
