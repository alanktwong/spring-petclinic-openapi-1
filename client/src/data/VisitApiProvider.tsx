import * as React from 'react';
import { VisitApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithVisitApiProps {
  visitApi: VisitApi;
}

export const VisitApiContext = React.createContext(new VisitApi());

export const withVisitApi = <P extends WithVisitApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithVisitApi extends React.Component<Subtract<P, WithVisitApiProps>> {
    static contextType = VisitApiContext;
    context!: React.ContextType<typeof VisitApiContext>;

    render() {
      return <Component visitApi={this.context} {...(this.props as P)} />;
    }
  };
