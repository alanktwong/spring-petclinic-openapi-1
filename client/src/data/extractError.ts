import { IError } from '../types';

const extractError = (response): Promise<IError> => {
  if (response.json) {
    return response.json();
  } else {
    return Promise.resolve({ message: `${response}` });
  }
};

export default extractError;
