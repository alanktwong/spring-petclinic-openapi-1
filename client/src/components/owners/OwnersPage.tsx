import * as React from 'react';

import { RouteComponentProps, withRouter } from 'react-router-dom';

import { Owner } from 'petclinic-api';
import { withOwnerApi, WithOwnerApiProps } from '../../data/OwnerApiProvider';

import OwnerInformation from './OwnerInformation';
import PetsTable from './PetsTable';

import { IError } from '../../types';
import PageErrorMessage from '../PageErrorMessage';
import extractError from '../../data/extractError';

import Loading from '../Loading';

interface IOwnersPageProps extends RouteComponentProps, WithOwnerApiProps {}

interface IOwnerPageState {
  error?: IError;
  owner?: Owner;
}

class OwnersPage extends React.Component<IOwnersPageProps, IOwnerPageState> {
  async componentDidMount() {
    const ownerId = Number(this.props.match.params['ownerId']);

    try {
      const owner = await this.props.ownerApi.getOwner({ ownerId });

      this.setState({ owner });
    } catch (response) {
      const error = await extractError(response);
      this.setState({ error });
    }
  }

  render() {
    const { error, owner } = this.state || {};

    if (error) {
      return <PageErrorMessage error={error} />;
    }

    if (!owner) {
      return <Loading />;
    }

    return (
      <div>
        <OwnerInformation owner={owner} />
        <PetsTable owner={owner} />
      </div>
    );
  }
}

export default withOwnerApi(withRouter(OwnersPage));
