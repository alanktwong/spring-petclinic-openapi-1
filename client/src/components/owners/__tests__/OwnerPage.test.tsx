import React from 'react';
import renderer from 'react-test-renderer';
import OwnersPage from '../OwnersPage';
import waitForExpect from 'wait-for-expect';
import { OwnerApiContext } from '../../../data/OwnerApiProvider';
import { OwnerApi } from 'petclinic-api';
import { Route, StaticRouter } from 'react-router-dom';

it('renders correctly', async () => {
  const mock = new OwnerApi();
  mock.getOwner = async () => ({
    id: 1,
    firstName: 'firstName1',
    lastName: 'lastName1',
    address: 'address1',
    city: 'city1',
    telephone: '012345',
    pets: [
      {
        id: 2,
        name: 'petName2',
        birthDate: new Date(2019, 0, 1),
        typeId: 3,
        type: {
          id: 3,
          name: 'typeName3',
        },
        visits: [
          {
            id: 4,
            date: new Date(2019, 1, 2),
            description: 'description4',
          },
        ],
      },
    ],
  });

  const component = renderer.create(
    <StaticRouter location="/owners/1">
      <OwnerApiContext.Provider value={mock}>
        <Route path="/owners/:ownerId(\d+)" component={OwnersPage} />
      </OwnerApiContext.Provider>
    </StaticRouter>
  );

  const page = component.root.find(node => node.type.name === 'OwnersPage');

  await waitForExpect(() => {
    expect(page.instance.state.owner.id).toBe(1);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
