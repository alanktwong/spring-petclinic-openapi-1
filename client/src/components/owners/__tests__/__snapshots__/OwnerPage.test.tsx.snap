// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <section>
    <h2>
      Owner Information
    </h2>
    <table
      className="table table-striped"
    >
      <tbody>
        <tr>
          <th>
            Name
          </th>
          <td>
            <b>
              firstName1
               
              lastName1
            </b>
          </td>
        </tr>
        <tr>
          <th>
            Address
          </th>
          <td>
            address1
          </td>
        </tr>
        <tr>
          <th>
            City
          </th>
          <td>
            city1
          </td>
        </tr>
        <tr>
          <th>
            Telephone
          </th>
          <td>
            012345
          </td>
        </tr>
      </tbody>
    </table>
    <a
      className="btn btn-default"
      href="/owners/1/edit"
      onClick={[Function]}
    >
      Edit Owner
    </a>
    Â 
    <a
      className="btn btn-default"
      href="/owners/1/pets/new"
      onClick={[Function]}
    >
      Add New Pet
    </a>
  </section>
  <section>
    <h2>
      Pets and Visits
    </h2>
    <table
      className="table table-striped"
    >
      <tbody>
        <tr>
          <td
            style={
              Object {
                "verticalAlign": "top",
              }
            }
          >
            <dl
              className="dl-horizontal"
            >
              <dt>
                Name
              </dt>
              <dd>
                petName2
              </dd>
              <dt>
                Birth Date
              </dt>
              <dd>
                2019-01-01
              </dd>
              <dt>
                Type
              </dt>
              <dd>
                typeName3
              </dd>
            </dl>
          </td>
          <td
            style={
              Object {
                "verticalAlign": "top",
              }
            }
          >
            <table
              className="table-condensed"
            >
              <thead>
                <tr>
                  <th>
                    Visit Date
                  </th>
                  <th>
                    Description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    2019-02-02
                  </td>
                  <td>
                    description4
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/owners/1/pets/2/edit"
                      onClick={[Function]}
                    >
                      Edit Pet
                    </a>
                  </td>
                  <td>
                    <a
                      href="/owners/1/pets/2/visits/new"
                      onClick={[Function]}
                    >
                      Add Visit
                    </a>
                  </td>
                </tr>
              </tbody>
            </table>
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</div>
`;
